File: mifosng-android/src/main/java/com/mifos/objects/client/Page.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.objects.client;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by ishankhanna on 09/02/14.
 */
public class Page<T> {

    private int totalFilteredRecords;
    private List<T> pageItems = new ArrayList<T>();


    public List<T> getPageItems() {
        return pageItems;
    }
    public void setPageItems(List<T> pageItems) {
        this.pageItems = pageItems;
    }


    public int getTotalFilteredRecords() {
        return totalFilteredRecords;
    }

    public void setTotalFilteredRecords(int totalFilteredRecords) {
        this.totalFilteredRecords = totalFilteredRecords;
    }


}

File: core/database/src/main/java/com/mifos/core/objects/client/Page.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.objects.client

/**
 * Created by ishankhanna on 09/02/14.
 */
data class Page<T> (
    var totalFilteredRecords: Int = 0,

    var pageItems: List<T> = ArrayList()
)